Assets/MixedRealityToolkit.Extensions/SpectatorView/Scripts/SpatialCoordinateSystemManager.cs

```diff
@@ -0,0 +1,123 @@
+        internal SpatialLocalizer spatialLocalizer;
+
+        public void OnConnected(SocketEndpoint endpoint)
+        {
+            if (members.ContainsKey(endpoint))
+            {
+                Debug.LogWarning("SpatialCoordinateSystemMember connected that already existed");
+                return;
+            }
+
+            if (spectatorView.Role == Role.Spectator)
+            {
+                if (members.Count > 0 &&
+                    !members.ContainsKey(endpoint))
+                {
+                    Debug.LogWarning("A second SpatialCoordinateSystemMember connected while the device was running as a spectator. This is an unexpected scenario.");
+                    return;
+                }
+            }
+
+            var roleOfConnection = (spectatorView.Role == Role.User) ? Role.Spectator : Role.User;
+            var member = new SpatialCoordinateSystemMember(Role.Spectator, endpoint, null, true);
+            members[endpoint] = member;
+            if (spatialLocalizer != null)
+            {
+                member.LocalizeAsync(spatialLocalizer).FireAndForget();
+            }
+        }
+
+        public void OnDisconnected(SocketEndpoint endpoint)
+        {
+            if (members.TryGetValue(endpoint, out var member))
+            {
+                member.Dispose();
+                members.Remove(endpoint);
+            }
+        }
+
+        public void HandleCommand(string command, SocketEndpoint endpoint, BinaryReader reader)
+        {
+            switch (command)
+            {
+                case SpatialLocalizationMessageHeader:
+                    {
+                        if (!members.TryGetValue(endpoint, out var member))
+                        {
+                            Debug.LogError("Received a message for an endpoint that had no associated spatial coordinate system member");
+                        }
+                        else
+                        {
+                            member.ReceiveMessage(reader);
+                        }
+                    }
+                    break;
+            }
+        }
+
+        protected override void Awake()
+        {
+            RegisterCommands();
+        }
+
+        protected override void OnDestroy()
+        {
+            UnregisterCommands();
+            CleanUpMembers();
+        }
+
+        private void RegisterCommands()
+        {
+            foreach(var command in supportedCommands)
+            {
+                StateSynchronizationBroadcaster.Instance.Register(command, this);
+                StateSynchronizationObserver.Instance.Register(command, this);
+            }
+        }
+
+        private void UnregisterCommands()
+        {
+            foreach (var command in supportedCommands)
+            {
+                StateSynchronizationBroadcaster.Instance.Unregister(command, this);
```

Comment:
>StateSynchronizationBroadcaster.Instance [](start = 16, length = 40)

These probably need null checks for cases where there's only one of the two in the scene? Or will they always both be there because of the unified prefab?